pkgs/applications/video/mplayer/default.nix

```diff
@@ -1,4 +1,4 @@
-{ config, lib, stdenv, fetchurl, fetchsvn, pkg-config, freetype, yasm, ffmpeg_4
+{ config, lib, stdenv, fetchurl, pkg-config, freetype, yasm, ffmpeg_4
```

Comment:
according to http://www.mplayerhq.hu/design7/news.html

compatible with ffmpeg 5

Reply:
we would need to introduce a patch to fix this. it should compile with 5.0 but it doesnt with 5.1.4 because of the following error

```
av_helpers.c: In function 'mp_msp_av_log_callback':
av_helpers.c:54:32: error: 'const struct AVCodec' has no member named 'decode'
   54 |                     if(s->codec->decode)
      |                                ^~
av_helpers.c:57:32: error: 'const struct AVCodec' has no member named 'decode'
   57 |                     if(s->codec->decode)
```

it seems like this decode function was removed (or moved, not sure yet) here: https://github.com/FFmpeg/FFmpeg/commit/20f972701806be20a77f808db332d9489343bb78#diff-e2ffbd866f8196aea62dcc198390f6f5778c4fd3d0e53e5ca35d57fee2305d94L315-L328

i'll check if there is a patch available to fix this issue

Reply:
i am fine with doing it here: compile error was fixed here svn://svn.mplayerhq.hu/mplayer/trunk@38361 i tried it and it seems to work